File: frontend/src/api/tts.ts
/** M5 TTS配音 - API服务层 */

import type { VoiceInfo, EmotionPreset, TTSJobRequest, TTSJobResult } from './ttsTypes'

const BASE = 'http://localhost:8000/api/v1/tts'

async function req<T>(url: string, opts?: RequestInit): Promise<T> {
  const res = await fetch(url, { headers: { 'Content-Type': 'application/json' }, ...opts })
  if (!res.ok) throw new Error(((await res.json().catch(() => ({}))).detail) || res.statusText)
  return res.json()
}

export async function getVoices(gender?: string, style?: string): Promise<VoiceInfo[]> {
  const p = new URLSearchParams()
  if (gender) p.set('gender', gender); if (style) p.set('style', style)
  return req(`${BASE}/voices?${p}`)
}

export async function getEmotionPresets(): Promise<EmotionPreset[]> {
  return req(`${BASE}/emotion-presets`)
}

export async function synthesize(req: TTSJobRequest): Promise<TTSJobResult> {
  return req(`${BASE}/synthesize`, { method: 'POST', body: JSON.stringify(req) })
}

File: backend/app/api/tts.py
"""M5 TTS配音服务 - API 路由"""

from __future__ import annotations
from fastapi import APIRouter, HTTPException, Query

from app.models.tts import (
    VoiceInfo, EmotionPreset, EmotionParams,
    TTSJobRequest, TTSJobResult, TTSJob,
)
from app.services.tts.engine import tts_engine

router = APIRouter()


# ---- 声线管理 ----

@router.get("/voices", response_model=list[VoiceInfo])
async def list_voices(gender: str | None = Query(None), style: str | None = Query(None)):
    """F5.1 声线列表"""
    return tts_engine.get_voices(gender, style)


@router.get("/voices/{voice_id}", response_model=VoiceInfo)
async def get_voice(voice_id: str):
    voice = tts_engine.get_voice(voice_id)
    if not voice:
        raise HTTPException(status_code=404, detail="声线不存在")
    return voice


# ---- 情感预设 ----

@router.get("/emotion-presets", response_model=list[EmotionPreset])
async def list_emotion_presets():
    """F5.2 情感预设列表"""
    return tts_engine.get_emotion_presets()


# ---- 配音任务 ----

@router.post("/synthesize", response_model=TTSJobResult)
async def synthesize(req: TTSJobRequest):
    """F5.3 批量分段配音"""
    if not req.segments:
        raise HTTPException(status_code=400, detail="段落列表不能为空")
    result = tts_engine.synthesize(req)
    return result


@router.get("/jobs/{job_id}", response_model=TTSJob)
async def get_job(job_id: str):
    job = tts_engine.get_job(job_id)
    if not job:
        raise HTTPException(status_code=404, detail="任务不存在")
    return job
